docs/modules/ROOT/pages/how-to/connect-gitlab.adoc

```diff
@@ -2,61 +2,174 @@
 
 image:logos/gitlab.svg[role="related thumb right",alt="GitLab logo",width=120,height=120] This page explains how to connect your GitLab project to {product} for your GitOps operations.
 
-This integration has two major aspects:
+This integration has three significant aspects:
 
-. Configuring {product} as a Kubernetes cluster for your GitLab project.
+. Configuring {product} as a Kubernetes cluster for your GitLab project *with* the Agent.
+. Configuring {product} as a Kubernetes cluster for your GitLab project *without* the Agent.
 . Connecting {product} to access your GitLab project's container registry.
 
 == Prerequisites
 
-For this guide, it's assumed that:
-
-* You have a GitLab project (either on https://gitlab.com or in a private instance).
-* You are logged in to your {product} project using the `oc login` command.
-
-== Configuring {product} as a Kubernetes cluster in GitLab
+For this guide, you need:
+
+* A GitLab project (either on https://gitlab.com or in a private instance).
+* Log in to your {product} project using the `oc login` command, and select your project.
+
+== Configuring {product} as a Kubernetes cluster in GitLab with the Agent
+
+To configure your {product} project as a Kubernetes cluster in GitLab using the new GitLab Agent, follow the steps outlined below.
+
+IMPORTANT: Given the shared nature of {product}, the GitLab Agent can't be installed cluster-wide, but it can be installed "organization-wide," however, as shown in this section.
+
+On your GitLab instance:
+
+. Create a file named `.gitlab/agents/appuio-cloud/config.yaml` in your GitLab project; commit and push.
+. In GitLab, select the menu:Infrastructure[Kubernetes Clusters] screen and click the btn:[Connect a cluster (agent)].
+. In the popup that opens, click on the menu:[Select an agent or enter a name to create new] menu and select the `appuio-cloud` entry, corresponding to the file created above.
+. Copy the GitLab Agent token shown in the popup, as it won't appear anymore.
+
+In your terminal, perform the following steps:
+
+. Add the GitLab repository to `helm` and update:
++
+[source, shell]
+--
+helm repo add gitlab https://charts.gitlab.io && helm repo update
+--
+
+. Use the GitLab Agent token in the following command:
++
+[source, shell]
+--
+helm upgrade --install gitlab-agent gitlab/gitlab-agent \
+    --set config.token={GITLAB_AGENT_TOKEN} \
+    --set config.kasAddress=wss://kas.gitlab.com \
+    --set rbac.create=false
+--
+
+NOTE: If you're using a self-hosted GitLab instance, change the value of the `config.kasAddress` parameter to match your own.
+
+. Grant namespace admin permissions for the Agent's service account:
++
+[source, shell]
+--
+oc policy add-role-to-user admin -z gitlab-agent --rolebinding-name gitlab-agent
+--
+
+You can now modify your `.gitlab-ci.yml` file to deploy your manifests whenever needed; for example, every time a new tag is pushed:
+
+[source, yaml]
```

Comment:
I usually put the filename as the title of the code snippet, but no hard feeling.
```suggestion

.gitlab-ci.yml
[source, yaml]
```